controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/WorkerPool.java

```diff
@@ -79,8 +93,8 @@ public boolean assignTask(TaskBase task) {
    * @param cluster the name of the cluster
    * @return
    */
-  public boolean abortTask(String cluster) throws Exception {
-    FutureTask runningTask = runningTasks.get(cluster);
+  public synchronized boolean abortTask(String cluster) throws Exception {
```

Comment:
There are two critical sections we synchronize on:
```
Future f = executorService.submit(...);
runningTasks.put(clusterName, f);
```
and
```
Future f = runningTasks.get(clusterName);
f.cancel(...);
```
We don't want two threads interleaving these two critical section.